models/stics/R/met2model.STICS.R

```diff
@@ -0,0 +1,183 @@
+                        formatname = "climate",
+                        startdate = start_date,
+                        enddate = end_date,
+                        dbfile.name = out.files,
+                        stringsAsFactors = FALSE)
+  PEcAn.logger::logger.info("internal results")
+  PEcAn.logger::logger.info(results)
+  
+  
+  ## check to see if the outfolder is defined, if not create directory for output
+  if (!file.exists(outfolder)) {
+    dir.create(outfolder)
+  }
+  
+  ctr <- 1
+  
+  ## loop over files/years
+  for (year in seq(start_year, end_year)) {
+    
+    if (file.exists(out.files.full[ctr]) && !overwrite) {
+      PEcAn.logger::logger.debug("File '", out.files.full[ctr], "' already exists, skipping to next file.")
+      next
+    }
+    
+    PEcAn.logger::logger.info(year)
+    
+    ## handle dates, also for partial year(s)
+    if(year == start_year & year != end_year){
+      # start year could be full or partial
+      simdays <- seq(lubridate::yday(start_date), PEcAn.utils::days_in_year(year))
+    }else if(year == end_year & year != start_year){
+      # end year could be full or partial
+      simdays <- seq(1, lubridate::yday(end_date))
+    }else if(year == end_year & year == start_year){
+      # we have one full or partial year
+      simdays <- seq(lubridate::yday(start_date), lubridate::yday(end_date))
+    }else{
+      # a full year in between
+      simdays <- seq(1, PEcAn.utils::days_in_year(year))
+    }
+    
+    
+    NDAYS      <- length(simdays)
+    NWEATHER   <- as.integer(13)
+    weather_df <- as.data.frame(matrix( -999.9, nrow = NDAYS, ncol = NWEATHER))
+    
+    # prepare data frame for STICS format, daily inputs, but doesn't have to be full year
+    weather_df[ ,1] <- rep(gsub(".*_STICS_site_", "", outfolder), NDAYS) # column 1: name of weather file
+    weather_df[ ,2] <- rep(year, NDAYS) # column 2: year
+    start_month <- ifelse(year == start_year, paste0(start_date), paste0(year, "/01/01"))
+    end_month   <- ifelse(year == end_year,   paste0(end_date),   paste0(year, "/12/31"))
+
+    weather_df[ ,3] <- lubridate::month(seq(lubridate::as_date(start_month), 
+                                            lubridate::as_date(end_month), by = "day")) # column 3: month
+    
+    weather_df[ ,4] <- lubridate::mday(seq(lubridate::as_date(start_month), 
+                          lubridate::as_date(end_month), by = "day")) # column 4: day in month
+    weather_df[ ,5] <- simdays # column 5: Julian day
+    
+    ## handle variables
+    old.file <- file.path(in.path, paste(in.prefix, year, "nc", sep = "."))
+    if (file.exists(old.file)) {
+      ## open netcdf
+      nc <- ncdf4::nc_open(old.file) 
+      on.exit(ncdf4::nc_close(nc), add = TRUE)
+      
+      ## convert time to seconds
+      sec <- nc$dim$time$vals
+      sec <- udunits2::ud.convert(sec, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
+      
+      dt <- PEcAn.utils::seconds_in_year(year) / length(sec)
```

Comment:
Does this work right if oldfile contains a partial year? If not, do we ensure elsewhere that oldfile is complete?

Reply:
tbh, I haven't error-proofed this initial coupling much, but one of my test cases is indeed with files that contain partial years. At least it runs through..but I need to check if it holds logically too